go/protocol/keybase1/extras.go

```diff
@@ -1619,6 +1619,98 @@ func (t TeamNamePart) Eq(t2 TeamNamePart) bool {
 	return string(t) == string(t2)
 }
 
+var implicitTeamPrefix = "__keybase_implicit_team__"
+var implicitSuffixLengthBytes = 16
+var implicitRxxString = fmt.Sprintf("^%s[0-9a-f]{%d}$", implicitTeamPrefix, implicitSuffixLengthBytes*2)
+var implicitNameRxx = regexp.MustCompile(implicitRxxString)
+
+func ImplicitTeamBackingNameFromString(s string) (ret ImplicitTeamBackingName, err error) {
+	if !implicitNameRxx.MatchString(s) {
+		return ret, fmt.Errorf("Bad implicit team name.")
+	}
+	return ImplicitTeamBackingName{
+		s[len(implicitTeamPrefix):],
+	}, nil
+}
+
+func (t ImplicitTeamBackingName) String() string {
+	return implicitTeamPrefix + t.Suffix
+}
+
+func (t ImplicitTeamBackingName) Eq(t2 ImplicitTeamBackingName) bool {
+	return t.String() == t2.String()
+}
+
+func (t ImplicitTeamBackingName) ToTeamID() TeamID {
+	low := strings.ToLower(t.String())
+	sum := sha256.Sum256([]byte(low))
+	bs := append(sum[:15], TEAMID_SUFFIX)
+	res, err := TeamIDFromString(hex.EncodeToString(bs))
+	if err != nil {
+		panic(err)
+	}
+	return res
+}
+
+func parseImplicitTeamPart(s string) (typ string, name string, err error) {
+	nColons := strings.Count(s, ":")
+	nAts := strings.Count(s, "@")
+	nDelimiters := nColons + nAts
+	if nDelimiters > 1 {
+		return "", "", fmt.Errorf("Invalid implicit team part, can have at most one ':' xor '@'")
+	}
+	if nDelimiters == 0 {
+		return "keybase", s, nil
+	}
+	if nColons == 1 {
+		parts := strings.Split(s, ":")
+		typ = parts[0]
+		name = parts[1]
+	} else {
+		parts := strings.Split(s, "@")
+		name = parts[0]
+		typ = parts[1]
+	}
+	return typ, name, nil
+}
+
+func ParseImplicitTeamTLFName(s string) (ImplicitTeamName, error) {
```

Comment:
can we move this code in `libkb` that way we can user the assertion parsing code we already have there?